sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpressions.scala

```diff
@@ -46,6 +46,38 @@ object GpuExpressionsUtils extends Arm {
       throw new IllegalStateException("Internal Error GPU support for this data type is not " +
         "implemented and should have been disabled")
   }
+
+  /**
+   * Tries to resolve a `GpuColumnVector` from a Scala `Any`.
+   *
+   * Now it covers only 3 types as below, and will blow up for other types.
+   *   - GpuColumnVector
+   *   - Scalar
+   *   - null
+   *
+   * This is a common handling of the result from the `columnarEval` of an expression.
+   * And it is recommended to return only one of the two types from a GPU expression's
+   * `columnarEval`.
+   *
+   * Not sure if `null` should be taken into account, since null should be replaced by
+   * `GpuScalar(null, dataType)` or a null ColumnVector. But what if some exceptions occur.
+   *
+   * @param any the input value. It will be closed if it is a closeable after the call done.
+   * @param numRows the expected row number of the output column, used when 'any' is a Scalar.
+   * @param dType the data type of the output column, used when 'any' is a Scalar.
+   * @return a `GpuColumnVector` if it succeeds. Users should close the column vector to avoid
+   *         memory leak.
+   */
+  def resolveColumnVector(any: Any, numRows: Int, dType: DataType): GpuColumnVector = {
+    withResourceIfAllowed(any) {
+      case c: GpuColumnVector => c.incRefCount()
+      case s: Scalar => GpuColumnVector.from(s, numRows, dType)
+      case null => GpuColumnVector.fromNull(numRows, dType)
+      case _ =>
+        throw new IllegalArgumentException(s"Cannot resolve a ColumnVector from the value: $any." +
+          s" Please convert the value to a Scalar or a GpuColumnVector before returning it.")
```

Comment:
I was thinking this would be called by _all_ cases where we're processing an expression result, and in that case we could seamlessly support expressions that return the Scala value.  I'm fine if we want to use this as the enforcement point for only allowing either `GpuColumnVector` or `GpuScalar` to be returned from `columnarEval`.

Reply:
Got it. Now updated again to stick to raising an exception for other cases here. 
Because we "don't want to support both Scalar and Scala Literal values being returned". (from https://github.com/NVIDIA/spark-rapids/pull/2313#pullrequestreview-650266129)